canals/component/component.py

```diff
@@ -15,6 +15,43 @@
 logger = logging.getLogger(__name__)
 
 
+# We ignore too-few-public-methods Pylint error as this is only meant to be
+# the definition of the Component interface.
+# A concrete Component will have more than method in any case.
+class Component(Protocol):  # pylint: disable=too-few-public-methods
```

Comment:
With a method and 4 properties PyLint still complains?? That's madness :exploding_head: You haven't wrote this before adding the properties, right?

Reply:
Ok I tested it, it's real. Wild.